### ISSUE TITLE
Flags `--interactive` and `--progress` do not play nicely with each other

### ISSUE THREAD

**Issue by arkban** [rclone/rclone#6755]
#### The associated forum post URL from `https://forum.rclone.org`

N/A, I am able to work around this issue by simply not using both `--interactive` and `--progress` flags at the same time.

#### What is the problem you are having with rclone?

I just recently started using `rclone sync` command. As mentioned by the [docs](https://rclone.org/commands/rclone_sync/) I tried using the `--interactive` and `--progress` flags. All behavior is working as expected; the issue is the output is...hard to read. (The "last" progress display works properly and still updates in place correctly.):

```
$ rclone sync --progress --interactive dropbox:/Books ./Books
Transferred:              0 B / 0 B, -, 0 B/s, ETA -
Checks:              3234 / 3234, 100%
Elapsed time:         7.1srclone: copy "Games/RPGs/D&D 5th Edition/Character Sheets/Pregen Characters 1-10/HumanPaladin1-10/Human Paladin 1.pdf"?
y) Yes, this is OK (default)
n) No, skip this
s) Skip all copy operations with no more questions
Transferred:              0 B / 4.029 MiB, 0%, 0 B/s, ETA -
Checks:              4423 / 4423, 100%
Transferred:            0 / 30, 0%
Elapsed time:        11.1s
Transferring:
 * Games/RPGs/D&D 5th Edi…10/Human Paladin 1.pdf: transferring
 * Games/RPGs/D&D 5th Edi…0/Human Paladin 10.pdf: transferring
 * Games/RPGs/D&D 5th Edi…10/Human Paladin 2.pdf: transferring
 * Games/RPGs/D&D 5th Edi…10/Human Paladin 3.pdf: transferring^C
```
_NOTE: The final `^C` is me manually stopping `rclone`

Further, the prompt is still responsive, e.g. hitting enter to trigger the default "Yes" answer gives me:

```
$ rclone sync --progress --interactive dropbox:/Books ./Books
Transferred:              0 B / 0 B, -, 0 B/s, ETA -
Checks:              2995 / 2995, 100%
Elapsed time:         6.7srclone: copy "Games/RPGs/D&D 5th Edition/Character Sheets/Pregen Characters 1-10/HumanPaladin1-10/Human Paladin 1.pdf"?
y) Yes, this is OK (default)
n) No, skip this
s) Skip all copy operations with no more questions
Transferred:              0 B / 7.520 MiB, 0%, 0 B/s, ETA -
Checks:              4656 / 4656, 100%
Transferred:            0 / 54, 0%
Elapsed time:        14.2s
Transferring:
 * Games/RPGs/D&D 5th Edi…10/Human Paladin 1.pdf: transferring
 * Games/RPGs/D&D 5th Edi…0/Human Paladin 10.pdf: transferring
Transferred:      130.059 KiB / 7.494 MiB, 2%, 12.996 KiB/s, ETA 9m40s
Checks:              4983 / 4983, 100%
Transferred:            1 / 54, 2%
Elapsed time:        17.2s
Transferring:
 * Games/RPGs/D&D 5th Edi…0/Human Paladin 10.pdf: transferring
 * Games/RPGs/D&D 5th Edi…10/Human Paladin 2.pdf: transferring
 * Games/RPGs/D&D 5th Edi…10/Human Paladin 3.pdf: transferring
 * Games/RPGs/D&D 5th Edi…10/Human Paladin 4.pdf: transferring
```

#### What is your rclone version (output from `rclone version`)

```
rclone v1.61.1
- os/version: darwin 13.1 (64 bit)
- os/kernel: 22.2.0 (arm64)
- os/type: darwin
- os/arch: arm64
- go/version: go1.19.4
- go/linking: dynamic
- go/tags: none
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

Mac OS X Ventura 13.1 (on Apple M1 Max)
```shell
$ uname -a
Darwin pbohan-<REDACTED> 22.2.0 Darwin Kernel Version 22.2.0: Fri Nov 11 02:03:51 PST 2022; root:xnu-8792.61.2~4/RELEASE_ARM64_T6000 arm64
```

#### Which cloud storage system are you using? (e.g. Google Drive)

Dropbox

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

```shell
$ rclone sync --progress --interactive dropbox:/Books ./Books
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

N/A, as the behavior is working as expected, only the UI experiences an issue. (If there is more data in the logs that I am unaware of please let me know and I will re-run with the `-vv` flag and post the output.)

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
What do you think we should do here?

1. Stop the progress output while waiting for user input?
2. Forbid `--interactive` and `--progress`?
3. Document how it works right now
4. Something else?

**Comment by arkban** [comment]
> What do you think we should do here?

I appreciate being asked! I think the solution is mostly just preventing the `--interactive` and `--progress` blocks from overwriting each other, i.e.
* There should be a link break between the end of `Elapsed time    N.Ns` and the `rclone: copy ..." prompt, i.e.
	```
    Elapsed time:         7.1srclone: copy "Games/RPGs/D&D 5th Edition/Character Sheets/Pregen Characters 
    ```
    should be
 	```
    Elapsed time:         7.1s
    rclone: copy "Games/RPGs/D&D 5th Edition/Character Sheets/Pregen Characters 
    ```
* There should be a line break between the user prompt and the "new" `Transferred..." line:
	```
	s) Skip all copy operations with no more questions
    Transferred:              0 B / 4.029 MiB, 0%, 0 B/s, ETA -
	```
	should be
	s) Skip all copy operations with no more questions
    y/n/s/> [answer]
    Transferred:              0 B / 4.029 MiB, 0%, 0 B/s, ETA -
    ```

I _think_ that would eliminate the confusion I saw.

(I admit I have no idea how difficult this might be, I am not familiar with curses-type output.)

> 1. Stop the progress output while waiting for user input?

I assumed `rclone` would continue to transferring other, non-conflicting data while waiting on the user's prompt. If it does pause everything, then that seems reasonable

> 2. Forbid --interactive and --progress?

That would definitely be the easiest and quickest fix! 😉 I can't say it's the "right" answer because I have no idea how often anyone uses the two flags in tandem.

> 3. Document how it works right now

Uhh, IMHO figuring out how to describe the interplay of these commands might be rather difficult. I can only show you what I happened to see.

> 4. Something else?

I think adding the line breaks would suffice.

**Comment by ncw** [comment]
> > What do you think we should do here?
> 
> I appreciate being asked! I think the solution is mostly just preventing the `--interactive` and `--progress` blocks from overwriting each other, i.e.
[snip]
> (I admit I have no idea how difficult this might be, I am not familiar with curses-type output.)

That would be super hard! The curses output isn't a plot this character here and this one here interface and rclone is doing all these things concurrently in different threads.

> > 1. Stop the progress output while waiting for user input?
> 
> I assumed `rclone` would continue to transferring other, non-conflicting data while waiting on the user's prompt. If it does pause everything, then that seems reasonable

I guess we could make rclone do either so stop the progress output and stop everything else progressing or let everything else progress and just don't write the progress output while we are waiting for user input.

> > 2. Forbid --interactive and --progress?
> 
> That would definitely be the easiest and quickest fix! wink I can't say it's the "right" answer because I have no idea how often anyone uses the two flags in tandem.

No me neither!

> > 3. Document how it works right now
> 
> Uhh, IMHO figuring out how to describe the interplay of these commands might be rather difficult. I can only show you what I happened to see.
> 
> > 4. Something else?
> 
> I think adding the line breaks would suffice.

I don't think that is an option alas.

So I think the favourite options are

 1. Stop the progress output while waiting for user input?
    - a.  pause everything while waiting for user input
    - b. keep running but just don't show progress

What do you prefer 1a or 1b?

1a is probably easiest and probably leads to the least surprises so that is what I'm leaning towards.

**Comment by arkban** [comment]
> > > What do you think we should do here?
> > 
> > 
> > I appreciate being asked! I think the solution is mostly just preventing the `--interactive` and `--progress` blocks from overwriting each other, i.e.
> > [snip]
> > (I admit I have no idea how difficult this might be, I am not familiar with curses-type output.)
> 
> That would be super hard! The curses output isn't a plot this character here and this one here interface and rclone is doing all these things concurrently in different threads.

I figured as much, hence my disclaimer attempt : )
 
> So I think the favourite options are
> 
>     1. Stop the progress output while waiting for user input?
>        
>        * a.  pause everything while waiting for user input
>        * b. keep running but just don't show progress
> 
> 
> What do you prefer 1a or 1b?
> 
> 1a is probably easiest and probably leads to the least surprises so that is what I'm leaning towards.

+1, applying the "least surprising" principle is probably the safest option.

**Comment by ncw** [comment]
Give this a go which should pause the progress output while interactively asking the user questions.

[v1.63.0-beta.6814.637775486.fix-6755-interactive-progress](https://beta.rclone.org/branch/fix-6755-interactive-progress/v1.63.0-beta.6814.637775486.fix-6755-interactive-progress/) on branch [fix-6755-interactive-progress](https://github.com/rclone/rclone/tree/fix-6755-interactive-progress) (uploaded in 15-30 mins)

This was a bit harder than I expected as it needed to avoid various deadlocks...

**Comment by micr0lab** [comment]
(For reference, this issue was also discussed at https://forum.rclone.org/t/progress-and-interactive-are-not-completely-compatible/28305 )
